File: instructor Henry/lenguajes/Python/orientaado-a-objetos/21-08-2024.py
class Transporte:
    def __init__(self, empresa="Carolina", ciudad="B/quilla"):
        self.empresa = empresa
        self.ciudad = ciudad

    def get_info(self):
        return f"{self.empresa}-{self.ciudad}"

class Publico(Transporte):
    def __init__(self, ninterno, empresa="Carolina", ciudad="B/quilla"):
        super().__init__(empresa, ciudad)
        self._ninterno = ninterno
        self.ruta = None

    def get_codigo(self):
        return f"{self.ruta} - {self._ninterno}"

class Particular(Transporte):
    def __init__(self, modelo, empresa="Carolina", ciudad="B/quilla"):
        super().__init__(empresa, ciudad)
        self.placa = None
        self.color = None
        self._modelo = modelo

    def get_info(self):
        return self.ciudad

    def get_rtm(self):
        return self._modelo + 5

def main():
    pu = Publico(920)
    pu.ruta = "Malambo City"
    print(pu.get_codigo())

    pub = Publico(450)
    pub.ruta = "Cano dulce"
    print(pub.get_codigo())

    print("------------------------------------")

    pa = Particular(2020)
    pa.placa = "XYZ 123"
    pa.color = "Rojo"
    print(pa.get_info())
    print(f"La técnico mecánica se hace el: {pa.get_rtm()}")

    print("")

    par = Particular(2024)
    par.placa = "XYZ 321"
    par.color = "Azul"
    print(par.get_info())
    print(f"La técnico mecánica se hace el: {par.get_rtm()}")

if __name__ == "__main__":
    main()

File: instructor Henry/lenguajes/Ruby/orientado-a-objetos/21-08-2024.rb
class Transporte
    def initialize(empresa = "Carolina", ciudad = "B/quilla")
      @empresa = empresa
      @ciudad = ciudad
    end
  
    def get_info
      "#{@empresa}-#{@ciudad}"
    end
  end
  
  class Publico < Transporte
    def initialize(ninterno, empresa = "Carolina", ciudad = "B/quilla")
      super(empresa, ciudad)
      @ninterno = ninterno
      @ruta = nil
    end
  
    def ruta=(ruta)
      @ruta = ruta
    end
  
    def get_codigo
      "#{@ruta} - #{@ninterno}"
    end
  end
  
  class Particular < Transporte
    def initialize(modelo, empresa = "Carolina", ciudad = "B/quilla")
      super(empresa, ciudad)
      @modelo = modelo
      @placa = nil
      @color = nil
    end
  
    def placa=(placa)
      @placa = placa
    end
  
    def color=(color)
      @color = color
    end
  
    def get_info
      @ciudad
    end
  
    def get_rtm
      @modelo + 5
    end
  end
  
  def main
    pu = Publico.new(920)
    pu.ruta = "Malambo City"
    puts pu.get_codigo
  
    pub = Publico.new(450)
    pub.ruta = "Cano dulce"
    puts pub.get_codigo
  
    puts "------------------------------------"
  
    pa = Particular.new(2020)
    pa.placa = "XYZ 123"
    pa.color = "Rojo"
    puts pa.get_info
    puts "La técnico mecánica se hace el: #{pa.get_rtm}"
  
    puts ""
  
    par = Particular.new(2024)
    par.placa = "XYZ 321"
    par.color = "Azul"
    puts par.get_info
    puts "La técnico mecánica se hace el: #{par.get_rtm}"
  end
  
  main
  